### ISSUE TITLE
0.8.0-beta.3 and 2.072 incompatibility 

### ISSUE THREAD

**Issue by bubnenkoff** [rejectedsoftware/vibe.d#1699]
Sorry that I upload files file here https://ufile.io/93cda
But it's take about 5 hours to understand why I can't run my project on VPS.
In result I understood that compiler 2.072 produce not working vibed project. It's run on my VPS but I can't open site. It's just show progress but does not happens.

By the link is two binaries builded with 2.072 and 2.073.1. The last one works fine.

I think it's better to add mention that vibed 0.8.0 require at last 2.073 compiler.

**Comment by SingingBush** [comment]
I updated a project to use vibe-0.8.0-beta.4 and was fine using dub with dmd on both Mac & Linux. Then tried `dub --compiler=ldc2` on the linux machine and got:

```bash
vibe-d:utils 0.8.0-beta.4: building configuration "library"...
../../../.dub/packages/vibe-d-0.8.0-beta.4/vibe-d/utils/vibe/internal/utilallocator.d(3,15): Error: module allocator is in file 'std/experimental/allocator.d' which cannot be read
```

Reason being is that I had the LDC version from Fedora repo which is pretty old (0.17.3) so had to uninstall it and download v1.1.1 instead. 

It's a pity that dub projects cannot [specify a minimum supported D language version](https://github.com/dlang/dub/issues/1097).

**Comment by s-ludwig** [comment]
The tests run on 2.072.0 up to 2.073.0, including HTTP server tests, that 2.072.0 isn't supported would be wrong to say per se, but maybe a word of warning is in order. @bubnenkoff: Just to make sure, do the exe files contain more code than what is in source/app.d, or is it possible to reproduce the issue by rebuilding the project as-is?

@SingingBush: True, that would be a very useful feature. The only reason why it's not included is mainly that managing such a compiler list manually is presumable prone to mistakes and bogus warnings when new compiler versions come out - and of course yet another thing that needs to be taken care of when a new library release is made. What would really be great would be to have CI integrated into code.dlang.org and have it test basic compatibility on its own (using a `dmd -o-` build for example).

**Comment by s-ludwig** [comment]
Okay, I could reproduce it locally, also with freshly rebuilt executables. Interestingly, replacing the "vibe-d" dependency with "vibe-http" fixes the issue. Based on the log output I strongly suspect it's https://issues.dlang.org/show_bug.cgi?id=16980 at play here. I have added a workaround within vibe.d for the cases that I stumbled over, but it looks like that is not enough.

Since the issue is already fixed, I'll just add a warning message to the repository README.
